### PR TITLE
Tidy up protobuf reserved fields.

### PR DESCRIPTION
**Pull request by pphaneuf**

### CODE REVIEW

File: trillian.proto [hunk 1]
```diff
@@ -251,6 +235,15 @@ message SignedLogRoot {
 
   // log_root_signature is the raw signature over log_root.
   bytes log_root_signature = 9;
+
+  reserved 1, 2, 3, 4, 5, 6, 7;
```
Comment (pavelkalinnikov) [inline trillian.proto:239]: Could use the `reserved 1 to 7;` syntax?

File: trillian.proto [hunk 2]
```diff
@@ -251,6 +235,15 @@ message SignedLogRoot {
...
+  reserved "key_hint";
```
Comment (pavelkalinnikov) [inline trillian.proto:240]: Optionally:
Add comments with int tags? E.g.:
```
  reserved "log_id"; // 5
```
Or sort by tag instead of lexicographically?

**Review by pavelkalinnikov — APPROVED** [review]
LGTM % optional nit

File: trillian.proto (hunk 1 above)
Comment (pphaneuf) [reply trillian.proto]: Ah yes, I always forget about that syntax! Done.

File: trillian.proto (hunk 2 above)
Comment (pphaneuf) [reply trillian.proto:240]: I don't think this really brings any useful information? The important thing is that these should no longer be used, and keeping them in lexicographical order is easiest to check and maintain.

Comment (pavelkalinnikov) [reply trillian.proto:240]: Fair enough.

**Pull request merged**